rspace/src/main/scala/coop/rchain/rspace/history/Trie.scala

```diff
@@ -2,10 +2,17 @@ package coop.rchain.rspace.history
 
 import coop.rchain.shared.AttemptOps._
 import scodec.Codec
-import scodec.bits.BitVector
+import scodec.bits.{BitVector, ByteVector}
 import scodec.codecs._
 
-sealed trait Trie[+K, +V]                         extends Product with Serializable
+sealed trait Child
+case class NodeChild(hash: Blake2b256Hash) extends Child
+case class LeafChild(hash: Blake2b256Hash) extends Child
+//case class Extension(affix: ByteVector, hash: Blake2b256Hash) extends Child
```

Comment:
Remove?

Also curious to hear your initial thoughts about modeling the type of Extension nodes.  My gut feeling is that they don't really fit here in the `Pointer` type.  Rather that they are another part of the `Trie` type...